### DISCUSSION TITLE
CN Tips about skywalking-java-agent development for Apple Silicon users

### DISCUSSION THREAD

**Discussion by Leibnizhu** [apache/skywalking#9293] [Show and tell]
I wrote some tips about skywalking-java-agent development for Apple Silicon users，in my blog (chinese): 

https://leibnizhu.github.io/p/Skywalking-Java-Agent%E5%9C%A8M1%E8%8A%AF%E7%89%87MacOS%E4%B8%8B%E7%9A%84%E5%BC%80%E5%8F%91tips/

But I am not so sure how to let other developers in skywalking community know about it.

**Comment by wu-sheng** [comment]
Actually, if you want, you could submit a pull request to enhance the document of plugin test, even more, you could use maven's `os-maven-plugin` to make the containers support MacOS AARM. This should be more helpful

**Reply by wu-sheng** [reply]
Another option is, whether this `createImageOptions` could be set through shell `--` parameter. If so, you could put these arguments on the test doc as a FAQ to guide users.
Once user doesn't have to change source codes to test on AARM, it would be good to host this solution as official.

**Reply by Leibnizhu** [reply]
thx, i would have a try.

**Comment by lujiajing1126** [comment]
Thanks for your tips. I think there are several steps we have to do in order to support `arm64/v8`,

1. Upgrade `protobuf`, `grpc` and related toolkits to have `aarm64` support,
2. Update `docker-maven-plugin` to build platform-specific variant of `tomcat` and `jvm` containers.

These are covered in your blog. But actually they are not sufficient, for example, 

1. For the `MySQL` scenarios, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/mysql-scenario/configuration.yml#L26, `mysql:5.7` does not have arm64/v8 support. An alternative may be `mariadb` images.
2. For the `Kafka` related cases, https://github.com/apache/skywalking-java/blob/main/test/plugin/scenarios/kafka-scenario/configuration.yml#L31, neither `zookeeper:3.4` nor `bitnami/kafka` images have arm64/v8 support. We have to replace them with `zookeeper:3.4.13` and `wurstmeister/kafka:2.12_2.2.1` respectively.
3. ... (pretty sure they must be other incompatible images and issues)

If we want to host all cases on arm64 machines,  may be we can create an issue to track.

**Reply by wu-sheng** [reply]
I think the key point of this discussion is making the test framework run on AARM, rather than all cases. That is not required. 
For example, MySQL and Zookeeper have versions not available on AARM, which is totally fine.
